app/controllers/communities_controller.rb

```diff
@@ -1,70 +1,98 @@
 class CommunitiesController < ApplicationController
-  before_action :authenticate_account!, except:  [ :index]
-  before_action :set_community, only: [:show, :edit, :update, :destroy, :mod, :check_if_banned]
+  before_action :authenticate_account!, except: [:index]
+  before_action :set_community, only: %i[show edit update destroy mod]
   before_action :community_list
   before_action :check_if_banned, only: [:show]
   after_action :count_post_for_this_week, only: [:index]
 
   def index
     count_post_for_this_week
     @categories = Community::CATEGORIES
-    if(params.has_key?(:category))
-      @communities = Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
-    else
-      @communities = Community.order(created_at: :desc).page(params[:page]).per 7
-    end
-    @random_category_communities = Community.where(category: @categories.sample).order("post_count_this_week desc").limit(5)
+    @communities = if params.has_key?(:category)
+                     Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
+                   else
+                     Community.order(created_at: :desc).page(params[:page]).per 7
+                   end
+    @random_category_communities = Community.where(category: @categories.sample).order('post_count_this_week desc').limit(5)
   end
 
   def show
-    @posts = @community.posts.limit(20).sort_by{ |p| p.score }.reverse
+    @community = Community.find(params[:id])
+    @posts = @community.posts.limit(20).sort_by { |p| p.score }.reverse
     @subscriber_count = @community.subscribers.count
-    @subscribed = is_subscribed?
+    @subscribed = if account_signed_in?
+                    Subscription.where(community_id: @community.id,
+                                       account_id: current_account.id).any?
+                  else
+                    false
+                  end
```

Comment:
see this? why is this happening?
Please make sure you don't mess up the code like this

Reply:
This is due to refactoring code using the RuboCop gem .I will resolve it now